qtism/data/content/xhtml/Img.php

```diff
@@ -194,79 +188,62 @@ public function hasLongdesc()
     }
 
     /**
-     * Set the height attribute. A negative (< 0) value for $height means there
-     * is no height indicated.
+     * Set the height of the image.
      *
-     * @param int|string $height An integer (pixels) or a string (percentage).
-     * @throws InvalidArgumentException If $height is not a valid integer or string value.
+     * A null value means that no height is provided.
+     *
+     * @param mixed $height A height.
+     * @throws InvalidArgumentException when $height is not valid.
      */
-    public function setHeight($height)
+    public function setHeight($height): void
     {
-        if ((is_int($height) && $height === -1) || Format::isXhtmlLength($height) === true) {
-            $this->height = $height;
-        } else {
-            $msg = "The 'height' argument must be a valid XHTML length value, '" . $height . "' given.";
-            throw new InvalidArgumentException($msg);
-        }
+        $this->height = Format::sanitizeXhtmlLength($height, 'height');
     }
 
     /**
-     * Get the height attribute. A negative (< 0) value for $height means there
-     * is no height indicated.
-     *
-     * @return int A height.
+     * Get the height of the image.
+     * Null means that no height is set.
      */
-    public function getHeight()
+    public function getHeight(): ?string
     {
         return $this->height;
     }
 
     /**
-     * Whether a height attribute is defined.
-     *
-     * @return bool
+     * Whether a height is set.
      */
-    public function hasHeight()
+    public function hasHeight(): bool
     {
-        return $this->getHeight() >= 0;
+        return $this->height !== null;
```

Comment:
Could it affect compiled deliveries with -1 set as default? 
e.g. it will return true but actually nothing been set

Reply:
Good catch! When we set -1, the value is set to null.